### ISSUE TITLE
Number of open issues

### ISSUE THREAD

**Issue by alecharp** [jenkins-infra/plugin-health-scoring#143]
Having the number of issues for a plugin can help understand the interaction between the users and the maintainers.

**Comment by alecharp** [comment]
Issues can be in GitHub or in Jira. Some plugins have specified that in their POM file and by doing so, it's advertised in the update-center. However, not all plugins have that.

I can create a probe which would count the issues and we would have a list of plugins which we need to attend to in order to improve their project configuration (adding the missing information in the POM file). As long as we don't use this metric in a scoring process, we won't be discriminating any plugin. WDYT @jleon33 ?

**Comment by dheerajodha** [comment]
For a given plugin, we can use update-center to understand which site it is using to track open issues, whether it is GH or JIRA or both GH and JIRA. I'm thinking about how would we fetch the open-issues count from these sites. We can get it easily from GH without any rate limit issues, but I'm curious about how we can get this info from JIRA.

Currently, we can use this JQL ([refer this](https://developer.atlassian.com/server/jira/platform/rest-apis/)) to get the open issues present within a component of a specific JIRA project: `https://host:port/context/rest/api/latest/search?jql=component=${component-number}%20AND%20status=open`. I've tried adding the required values to this URL and I'm able to get the count of open issues for a given plugin. But the rate limits of JIRA API are quite strict, [it says](https://developer.atlassian.com/server/hipchat/hipchat-rest-api-rate-limits/#rate-limits) "Your add-on can make 500 API requests per 5 minutes", which is not sufficient for us.

Wdyt?

**Comment by alecharp** [comment]
> We can get it easily from GH without any rate limit issues

I'm not sure. We are already using a lot of calls to GH but we will see.

About Jira, I'm not sure this documentation is about the Jenkins instance but about the cloud hosted ones (`*.atlassian.com`). But I haven't look at that for the moment. Give it a try.

**Comment by dheerajodha** [comment]
> Give it a try.

you mean, to try looking at the jira docs for the Jenkins instance? I don't know where to find that.
Or are you suggesting start working on this issue and then we'll figure out these API rate limit issues later when we hit them in the future?

**Comment by alecharp** [comment]
yes, try to work on the probe so we can see if there is a limitation. If so, we will see how we can work on it.

**Comment by Jagrutiti** [comment]
We can integrate Jenkins with JIRA: https://support.atlassian.com/jira-cloud-administration/docs/integrate-with-jenkins/

**Comment by Jagrutiti** [comment]
We can use [JQL](https://support.atlassian.com/jira-software-cloud/docs/jql-keywords/) and create something like the first example in the doc: `project = "New office" and status = "open"`.

**Comment by Jagrutiti** [comment]
I got the count for our repository using GitHub API: https://api.github.com/search/issues?q=repo:jenkins-infra/plugin-health-scoring+type:issue+state:open

Are we using this one for GitHub?

**Comment by hervelemeur** [comment]
Instead of giving an additional app access to Jira, another possibility could be to extend and use the data gathered by https://github.com/jenkins-infra/infra-reports and/or https://github.com/jenkins-infra/plugin-site-api, WDYT?

**Comment by Jagrutiti** [comment]
Thanks for sharing @hervelemeur 

`infra-reports` did help.

**Comment by Jagrutiti** [comment]
I see this API being called: `https://issues.jenkins.io/rest/api/2/group/member?groupname=jira-users`

in https://github.com/jenkins-infra/infra-reports/blob/main/jira-users-report/user-report.sh but cannot find a documentation.

**Comment by hervelemeur** [comment]
What would you like to know about this API call?

**Comment by Jagrutiti** [comment]
The `infra-reports` already returns jira users: https://github.com/jenkins-infra/infra-reports/blob/main/jira-users-report/user-report.sh

It invokes this API: https://issues.jenkins.io/rest/api/2/group/member?groupname=jira-users

But I am not able to find any documentation on this API.

I was wondering if I can tweak this API and return number of open issues in JIRA based on `component`.

The `issueTracker` in update-center gives the following information about a plugin that uses JIRA.

```
 "issueTrackers": [
        {
          "reportUrl": "https://www.jenkins.io/participate/report-issue/redirect/#27424",
          "type": "jira",
          "viewUrl": "https://issues.jenkins.io/issues/?jql=component=27424"
        }
      ],
```

**Comment by Jagrutiti** [comment]
This dashboard: https://issues.jenkins.io/secure/Dashboard.jspa

Does display all JIRA issues with their count. I was hoping to fetch the total in my project.

**Comment by Jagrutiti** [comment]
I can use this documentation: https://developer.atlassian.com/server/jira/platform/rest-apis/

With the following syntax? 
`http://host:port/context/rest/api-name/api-version/resource-name`

The `http://host:port/context/` in our case would be `https://issues.jenkins.io/`?
